sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxyUtil.java

```diff
@@ -0,0 +1,102 @@
+package com.azure.core.http.rest;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.CookiePolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.core.implementation.http.UnexpectedExceptionInformation;
+import com.azure.core.util.Context;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.SerializerAdapter;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RestProxyUtil {
+
+    static Context mergeRequestOptionsContext(Context context, RequestOptions options) {
+        if (options == null) {
+            return context;
+        }
+
+        Context optionsContext = options.getContext();
+        if (optionsContext != null && optionsContext != Context.NONE) {
+            context = CoreUtils.mergeContexts(context, optionsContext);
+        }
+
+        return context;
+    }
+
+    @SuppressWarnings("deprecation")
+    static void validateResumeOperationIsNotPresent(Method method, ClientLogger logger) {
+        // Use the fully-qualified class name as javac will throw deprecation warnings on imports when the class is
+        // marked as deprecated.
+        if (method.isAnnotationPresent(com.azure.core.annotation.ResumeOperation.class)) {
+            throw logger.logExceptionAsError(new IllegalStateException("'ResumeOperation' isn't supported."));
+        }
+    }
+
+    static Exception instantiateUnexpectedException(final UnexpectedExceptionInformation exception,
```

Comment:
You may want to pull in the changes that have been made in main for this, otherwise we're going to see either some nasty merge conflicts or accidental reverting of changes.